### ISSUE TITLE
Number of points for Multipoint always returns 0

### ISSUE THREAD

**Issue by evetion** [yeesian/ArchGDAL.jl#30]
```julia
julia> using ArchGDAL
julia> mp = ArchGDAL.createmultipoint([
                (1251243.7361610543, 598078.7958668759),
                (1240605.8570339603, 601778.9277371694),
                (1250318.7031934808, 606404.0925750365)
                ])
Geometry: MULTIPOINT (1251243.73616105 598078.795866876,1251 ... 5036)

julia> ArchGDAL.npoint(mp)
0
```

**Comment by yeesian** [comment]
We have the following situation:
```julia
julia> AG.npoint(geom)
0

julia> AG.ngeom(geom)
4
```

I can see arguments going both ways... How do you think we should handle this?

# Explanation
[`AG.npoint`](http://gdal.org/1.11/ogr/ogr__api_8h.html#aec43cbe83609bc7b43739d67858846c1) maps to [`OGR_G_GetPointCount`](http://www.gdal.org/ogr__api_8h.html#aec43cbe83609bc7b43739d67858846c1) which has the docstring:

> Fetch number of points from a geometry.
> 
> Only wkbPoint[25D] or wkbLineString[25D] may return a valid value. Other geometry types will silently return 0.

On the other hand, `AG.ngeom` maps to [`OGR_G_GetGeometryCount`](http://www.gdal.org/ogr__api_8h.html#a1fa07ddf969f97f6444de6ae5128d842) which has the following docstring:

> Fetch the number of elements in a geometry or number of geometries in container.
> 
> Only geometries of type wkbPolygon[25D], wkbMultiPoint[25D], wkbMultiLineString[25D], wkbMultiPolygon[25D] or wkbGeometryCollection[25D] may return a valid value. Other geometry types will silently return 0.

**Comment by yeesian** [comment]
This being `ArchGDAL`, I currently lean towards the current behavior, and leave a info/warning/error-message telling users to switch to `ngeom` if they're using `npoint`.

On the other hand, I am partial to sufficiently compelling arguments that we should consider our own definitions of `npoint` and `ngeom` that might depart from those of GDAL's.

**Comment by evetion** [comment]
I see ArchGDAL definitions as departing from GDAL (not doing that is GDAL.jl territory). Surely not in every aspect, but only to make it Julian. This can be done with new functions. So instead of editing ngeom or npoint I would just propose:

```julia
julia> length(mp)
3
```
for all collections (using `ngeom` on Multi collections, `npoint` on LineStrings).

I'm not sure why GDAL defines npoint for wkbPoint?

**Comment by rouault** [comment]
> is there a reason why OGR_G_GetGeometryCount doesn't use the definition of OGR_G_GetPointCount for wkbPoint[25D] or wkbLineString[25D]?

This come from the SQL/ML abstraction http://portal.opengeospatial.org/files/?artifact_id=25354 that define ST_NumGeometries() only on geometrycollection and derived subclasses.

**Comment by yeesian** [comment]
> This come from the SQL/ML abstraction http://portal.opengeospatial.org/files/?artifact_id=25354 that define ST_NumGeometries() only on geometrycollection and derived subclasses.

Thank you, that was helpful! 

---

Maarten, I don't think there's a real need for a 3rd method for now. Let's remove [`npoint`](https://github.com/yeesian/ArchGDAL.jl/blob/1855c55215c3b94d16df69a0a3829789775a1e59/src/ogr/geometry.jl#L720-L721) and use the following definition

```
"The number of elements in a geometry or number of geometries in container."
function ngeom(geom::AbstractGeometry)
    n = GDAL.getpointcount(geom.ptr)
    n == 0 ? GDAL.getgeometrycount(geom.ptr) : n
end
```

in place of the current [`ngeom`](https://github.com/yeesian/ArchGDAL.jl/blob/1855c55215c3b94d16df69a0a3829789775a1e59/src/ogr/geometry.jl#L862-L863). We can then replace all occurrences of `npoint` with `ngeom`.

**Comment by yeesian** [comment]
closed by #29
